### ISSUE TITLE
Options menu crashes client

### ISSUE THREAD

**Issue by robertdeclaux** [ZaneDubya/UltimaXNA#228]
Just click on options client crashes.

**Comment by ZaneDubya** [comment]
@robertdeclaux Would you post the error message associated with this bug?

**Comment by robertdeclaux** [comment]
Sorry i forgot to paste.
Error 2015-07-25 19: 06: 26: 5709 Index was out of range. It should not be a negative value and must be smaller than the size of collection.
Parameter name: index

Error 2015-07-25 19:06:26:5709 Dizin aralık dışındaydı. Negatif bir değer olmamalı ve koleksiyonun boyutundan daha küçük olmalıdır.
Parametre adı: index

**Comment by ZaneDubya** [comment]
@robertdeclaux Thank you - do you know what source file and line of code that error arose from?

**Comment by robertdeclaux** [comment]
DropDownList.cs at line 101, yesterday it wasn't happening.

**Comment by ZaneDubya** [comment]
I just submitted a fix. Try it after f459aa7b6540d0f6acceaeb6d451b98220b0f9ba? Also try selecting the first and last items in the drop down list to make sure it is saving the results of your selections. Thank you!

**Comment by robertdeclaux** [comment]
Still same error but which drop down list you mean, when i click on options button on paperdoll, it immediatelly crashes without showing the options gump

**Comment by ZaneDubya** [comment]
Hmm. Not sure what's causing this. Is there a longer error message? Check your debug.log or post a screen shot?

I was talking about the drop down list in the options menu, which you can't access. yet. ;)

**Comment by PatrikSamuelTauchim** [comment]
I cannot select any value from DropDownList now, it always return "Click here" so Index = -1. After reverting changes from #f459aa7

I was able to reproduce the issue. It will crash only if resolution is changed.
*\* Open Options
*\* Change resolution (does not matter to which one)
*\* Okay
*\* Open Options => Crash

```
System.ArgumentOutOfRangeException was unhandled
  HResult=-2146233086
  Message=Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
  Source=mscorlib
  ParamName=index
  StackTrace:
       at System.ThrowHelper.ThrowArgumentOutOfRangeException()
       at System.Collections.Generic.List`1.get_Item(Int32 index)
       at UltimaXNA.Ultima.UI.Controls.DropDownList.Update(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\UI\Controls\DropDownList.cs:line 104
       at UltimaXNA.Core.UI.AControl.UpdateControls(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Core\UI\AControl.cs:line 438
       at UltimaXNA.Core.UI.AControl.Update(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Core\UI\AControl.cs:line 369
       at UltimaXNA.Ultima.UI.Gump.Update(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\UI\Gump.cs:line 90
       at UltimaXNA.Ultima.UI.WorldGumps.OptionsGump.Update(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Ultima\UI\WorldGumps\OptionsGump.cs:line 162
       at UltimaXNA.Core.UI.UserInterfaceService.Update(Double totalMS, Double frameMS) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Core\UI\UserInterfaceService.cs:line 204
       at UltimaXNA.UltimaGame.Update(GameTime gameTime) in C:\Users\VM\Desktop\UltimaXNA-master\dev\UltimaGame.cs:line 198
       at Microsoft.Xna.Framework.Game.Tick()
       at Microsoft.Xna.Framework.Game.HostIdle(Object sender, EventArgs e)
       at Microsoft.Xna.Framework.GameHost.OnIdle()
       at Microsoft.Xna.Framework.WindowsGameHost.RunOneFrame()
       at Microsoft.Xna.Framework.WindowsGameHost.ApplicationIdle(Object sender, EventArgs e)
       at System.Windows.Forms.Application.ThreadContext.System.Windows.Forms.UnsafeNativeMethods.IMsoComponent.FDoIdle(Int32 grfidlef)
       at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
       at System.Windows.Forms.Application.Run(Form mainForm)
       at Microsoft.Xna.Framework.WindowsGameHost.Run()
       at Microsoft.Xna.Framework.Game.RunGame(Boolean useBlockingRun)
       at Microsoft.Xna.Framework.Game.Run()
       at UltimaXNA.Bootstrapper.StartEngine() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 79
       at UltimaXNA.Bootstrapper.Initialize() in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 62
       at UltimaXNA.Bootstrapper.Main(String[] args) in C:\Users\VM\Desktop\UltimaXNA-master\dev\Bootstrapper.cs:line 29
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 

```

**Comment by ZaneDubya** [comment]
@Eskymak @robertdeclaux so bizarre. Not crashing over here...

**Comment by ZaneDubya** [comment]
@Eskymak Ah, I see what the problem is now! Thank you for giving me the test case, that showed me what to do. Try updating again, it should be fixed.

**Comment by PatrikSamuelTauchim** [comment]
@ZaneDubya Not crashing anymore!

Sometimes, after changing gump resolution, window resolution dropdown shows Click here. Dont have test case for now

**Comment by ZaneDubya** [comment]
@Eskymak I've added the necessary hook ups for changing the play window size - it's really looking good. Thanks for all your work on the options gump :+1:

**Comment by PatrikSamuelTauchim** [comment]
@ZaneDubya Good job! Resizing of play window (gump) works! Can you offer  same resolutions in play window dropdown as for window resolution (supported ones)? (not just 640_480/800_600/1024*768)? Or should I look at that? (and mess-up with your code :-D)

**Comment by ZaneDubya** [comment]
Haha, can't offer all of them. The max dimension of the play window is 2048 in any direction, so attempting to create a play window with a resolution of (for example) 2560x1600 would crash the client immediately.

If you want to give it a go, be my guest - but remember to filter out ones that are too big in the x/y dimension! That said, I wonder if there's a better way to create a list of possible play window sizes. What if I wanted to play with a square play window? Or a window that is taller than it is wide? Or am I just being silly? :)

**Comment by PatrikSamuelTauchim** [comment]
@ZaneDubya I will filter out resolution with too big width and/or height, that should be fine.

I can also provide two text input fields (can we limit it to numbers only?) for custom resolution, dedicated just to silly players not satisfied with standardized resolutions. :-D

There is definitely better way on how to make that list, than my code put together from various internet findings :-)

**Comment by PatrikSamuelTauchim** [comment]
@ZaneDubya  Is there specific reason why you removed and commented out rest of the Update method? (values next to sliders are not updated) If not, I will put i back. :-)

PlayWindow resolution list implemented 9638607b949aa13b523cbc0243e2176e04678d20, custom resolution fields in next round.

**Comment by ZaneDubya** [comment]
Check the new way I am updating things in status gump. Much better way of doing things. :)

-Zane
